source/Calamari/Util/CrossPlatformExtensions.cs

```diff
@@ -16,11 +16,17 @@ public static string GetApplicationTempDir()
 #if NET40
             var path = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
 #else
-            var path = Environment.GetEnvironmentVariable("LOCALAPPDATA") ?? Environment.GetEnvironmentVariable("TMPDIR") ?? "/tmp";
+            var path = Environment.GetEnvironmentVariable("LOCALAPPDATA")
+                   ?? Environment.GetEnvironmentVariable("TMPDIR")
+                   ?? Environment.GetEnvironmentVariable("TEMP")
+                   ?? "/tmp";
 #endif
-            path = Path.Combine(path, Assembly.GetEntryAssembly().GetName().Name);
-            path = Path.Combine(path, "Temp");
-            return path;
+            if (Assembly.GetEntryAssembly() != null)
```

Comment:
GetApplicationTempDir() was existing but the first time used in tests, and was failing on the build server because how the test runs.

Do you mean the `?:` instead of `if-else`? It doesn't read as well... If there is a assembly name is there based on how it's running then it will work as it was originally written, the else case covers the case where team city is running it, and just another temp directory name will be fine.